redfish-core/lib/pcie.hpp

```diff
@@ -91,6 +92,206 @@ static inline void
+                    if (ec)
+                    {
+                        BMCWEB_LOG_DEBUG << "DBUS response error " << ec;
+                        messages::internalError(asyncResp->res);
+                        return;
+                    }
+
+                    const std::string* value =
+                        std::get_if<std::string>(&property);
+                    if (value == nullptr)
+                    {
+                        messages::internalError(asyncResp->res);
+                        return;
+                    }
+
+                    std::string linkStatus = *value;
+                    if (!linkStatus.empty())
+                    {
+                        fillPcieDeviceStatus(asyncResp->res, linkStatus);
+                        return;
+                    }
+                },
+                service, pcieSlotPath, "org.freedesktop.DBus.Properties", "Get",
+                interface, "LinkStatus");
+        }
+    }
+}
+
+/**
+ * @brief Get subtree map for PCIeSlots.
+ * @param[in,out]   asyncResp       Async HTTP response.
+ * @param[in]       pcieSlotPath    Object path of the PCIeSlot.
+ * @param[in]       pcieDevice      PCIe device name/ID.
+ */
+static inline void
+    getPcieSlotSubTree(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                       const std::string& pcieSlotPath,
+                       const std::string& pcieDevice)
+{
+    auto respHandler = [asyncResp, pcieSlotPath,
+                        pcieDevice](const boost::system::error_code ec,
+                                    const MapperGetSubTreeResponse& subTree) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR << "DBUS response error on GetSubTree"
+                             << ec.message();
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        if (subTree.size() == 0)
+        {
+            BMCWEB_LOG_ERROR << "Can't find PCIeSlot D-Bus object!";
+            return;
+        }
+
+        for (const auto& [objectPath, serviceMap] : subTree)
+        {
+            if (objectPath.empty() || serviceMap.size() != 1)
+            {
+                BMCWEB_LOG_ERROR << "Error getting PCIeSlot D-Bus object!";
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            if (pcieSlotPath != objectPath)
+            {
+                continue;
+            }
+
+            getPcieSlotLinkStatus(asyncResp, pcieSlotPath, serviceMap);
+            return;
+        }
+        BMCWEB_LOG_ERROR << "PCIe Slot not found for " << pcieDevice;
+    };
+
+    crow::connections::systemBus->async_method_call(
+        respHandler, "xyz.openbmc_project.ObjectMapper",
+        "/xyz/openbmc_project/object_mapper",
+        "xyz.openbmc_project.ObjectMapper", "GetSubTree",
+        "/xyz/openbmc_project/inventory", 0,
+        std::array<const char*, 1>{
+            "xyz.openbmc_project.Inventory.Item.PCIeSlot"});
```

Comment:
Nit: 
Can we make a string for this interface like we do for pcieDeviceInterface and use it here in this call?
static constexpr char const* pcieSlotInterface = "xyz.openbmc_project.Inventory.Item.PCIeSlot";